packages/contract-wrappers/src/contract_wrappers/erc20_proxy_wrapper.ts

```diff
@@ -1,38 +1,39 @@
+import { ERC20ProxyContract } from '@0xproject/abi-gen-wrappers';
+import { ERC20Proxy } from '@0xproject/contract-artifacts';
 import { AssetProxyId } from '@0xproject/types';
 import { Web3Wrapper } from '@0xproject/web3-wrapper';
 import { ContractAbi } from 'ethereum-types';
 import * as _ from 'lodash';
 
-import { artifacts } from '../artifacts';
 import { assert } from '../utils/assert';
 
 import { ContractWrapper } from './contract_wrapper';
-import { ERC20ProxyContract } from './generated/erc20_proxy';
 
 /**
  * This class includes the functionality related to interacting with the ERC20Proxy contract.
  */
 export class ERC20ProxyWrapper extends ContractWrapper {
-    public abi: ContractAbi = artifacts.ERC20Proxy.compilerOutput.abi;
+    public abi: ContractAbi = ERC20Proxy.compilerOutput.abi;
+    public address: string;
     private _erc20ProxyContractIfExists?: ERC20ProxyContract;
-    private _contractAddressIfExists?: string;
     /**
      * Instantiate ERC20ProxyWrapper
      * @param web3Wrapper Web3Wrapper instance to use
      * @param networkId Desired networkId
-     * @param contractAddressIfExists The contract address to use. This is usually pulled from
-     * the artifacts but needs to be specified when using with your own custom testnet.
+     * @param address (Optional) The address of the ERC20Proxy contract. If
+     * undefined, will default to the known address corresponding to the
+     * networkId.
      */
-    constructor(web3Wrapper: Web3Wrapper, networkId: number, contractAddressIfExists?: string) {
+    constructor(web3Wrapper: Web3Wrapper, networkId: number, address?: string) {
         super(web3Wrapper, networkId);
-        this._contractAddressIfExists = contractAddressIfExists;
+        this.address = _.isUndefined(address) ? this._getDefaultContractAddresses().erc20Proxy : address;
```

Comment:
I think we should wrap the call to `_getDefaultContractAddresses` in a try/catch and re-throw a more situation specific error message here. Perhaps: 

```
throw new Error(`'address' must be supplied to ERC20ProxyWrapper when instantiated with an unrecognized networkId (${networks.join(',')})`);
```

I find just allowing the base error to bubble up can be a source of a lot of confusion since it doesn't directly relate to what the user is currently doing.

Reply:
@fabioberger Good idea. I gathered your feedback here and from other comments and added some additional logic. See if you like the way I handle this error in [the latest commit](https://github.com/0xProject/0x-monorepo/pull/1105/commits/ae94f8cdb3b15209f35517c32cf03c278b68db80).